app/views/apipie_dsl/apipie_dsls/help.html.erb

```diff
@@ -1,58 +1,340 @@
+
+<%= apipie_dsl_example("#{apipie_erb_wrap('line1', close_trim: true)}\n#{apipie_erb_wrap('line2', close_trim: true)}", "line1line2") %>
+
+<p><%= _('This is used to reduce the number of lines, where Ruby syntax permits, in rendered templates. White spaces in ERB tags are ignored.') %></p>
+
+<p><%= _('The following example demonstrates how this is used to to remove unnecessary newlines between a FQDN and IP address:') %></p>
+
+<%= apipie_dsl_example("FQDN: #{apipie_erb_wrap('host.fqdn', close_trim: true)}\n IP: #{apipie_erb_wrap('host.ip', close_trim: true)}", "FQDN: host.example.com IP: 192.168.22.1") %>
+
+<p><%= _('Note the extra space before IP, the <code>-&#37;></code> removes the new line character only if there is no other text after it. Therefore the white space needs to be at the new line.').html_safe %></p>
+
+<h3 id="erb_comments"><%%# %></h3>
+
+<p><%= _('Encloses a comment that is ignored during template rendering:') %></p>
+
+<%= apipie_dsl_example("#{apipie_erb_wrap('A comment', mode: :comment, close_trim: true)}") %>
+
+<p><%= _('This generates no output. However if the closing tag was <code>&#37;></code> without the dash, it would generate a new line').html_safe %></p>
+
+<h3 id="erb_indentation"><%= _('Indentation in ERB templates') %></h3>
+
+<p><%= _('Because of the varying lengths of the ERB tags, indenting the ERB syntax might seem messy. ERB syntax ignores white space. One method of handling the indentation is to declare the ERB tag at the beginning of each new line and then use white space within the ERB tag to outline the relationships within the syntax, for example:') %></p>
+
+<pre>
+<&#37;- load_hosts.each_record do |host| -&#37;>
+<&#37;-   if host.build? &#37;>
+<&#37;=     host.name &#37;> build is in progress
+<&#37;-   end &#37;>
+<&#37;- end &#37;>
+</pre>
+
+<h2 id="building_blocks"><%= _('Basic building blocks - macros, variables, methods') %></h2>
+
+<p><%= _('The Ruby code inside of the ERB tags typically contain expressions. They consist of program flow control terminals, variables, macros. Variables and macros evaluates to objects. Each object has set of allowed methods that can be called on it. These methods can return objects.') %></p>
+<p><%= (_('An example of a control flow terminal is an if condition or for each cycle. Macros are functions called by their name without specifying any object, e.g. load_hosts or input. See all available macros %s. Example of object is a number, string, Host, Subnet, User. Calling a method on an object is denoted by <code>.</code>. For example <code>@host.fqdn</code> expression is an instance variable that holds and evaluates to an object of a Host, on which the fqdn method is called. To find the lits of all available methods on objects, see the documentation for each template type.') % link_to(_('here'), @doc[:doc_url] + section_ext('all'))).html_safe %></p>
```

Comment:
I would give here an example of code,  and on that code, I would mention what is an object/Macros/instance variable

Reply:
added with links for better explanation